hs_geographic_feature_resource/receivers.py

```diff
@@ -1,364 +1,116 @@
-                                                      "eastlimit":
-                                                      parsed_md_dict
-                                                      ["origin_extent_dict"]["eastlimit"],
-                                                      "projection_string":
-                                                      parsed_md_dict
-                                                      ["origin_projection_string"],
-                                                      "projection_name":
-                                                      parsed_md_dict["origin_projection_name"],
-                                                      "datum": parsed_md_dict["origin_datum"],
-                                                      "unit": parsed_md_dict["origin_unit"]
-                                                      }
-        metadata_array.append(original_coverage_dict)
-        metadata_dict["originalcoverage"] = original_coverage_dict
-
-        # field
-        field_info_array = []
-        field_name_list = parsed_md_dict["field_meta_dict"]['field_list']
-        for field_name in field_name_list:
-            field_info_dict_item = {}
-            field_info_dict_item['fieldinformation'] = \
-                parsed_md_dict["field_meta_dict"]["field_attr_dict"][field_name]
-            field_info_array.append(field_info_dict_item)
-        metadata_array.extend(field_info_array)
-        metadata_dict['field_info_array'] = field_info_array
-
-        # geometry
-        geometryinformation = {"featureCount": parsed_md_dict["feature_count"],
-                               "geometryType": parsed_md_dict["geometry_type"]}
-        metadata_array.append({"geometryinformation": geometryinformation})
-        metadata_dict["geometryinformation"] = geometryinformation
-        return(metadata_array, metadata_dict)
-    except:
-        raise Exception("Parse Shapefiles Failed!")
-
-
-# receiver used to extract metadata after user click on "create resource"
-@receiver(pre_create_resource, sender=GeographicFeatureResource)
-def geofeature_pre_create_resource(sender, **kwargs):
-
-    tmp_dir = None
+    resource = kwargs['resource']
+    res_file = None
+    base_err_msg = "Files failed validation ({reason}). Files not uploaded are: {files}."
+    files_failed_validation = []
+    failure_reasons = []
+    if resource.files.all().count() == 0:
+        # resource was created without any uploaded files - no file validation is needed
+        return
+    elif resource.files.all().count() == 1:
+        # this file has to be a zip file
+        res_file = resource.files.all().first()
+        if res_file.extension != '.zip':
+            files_failed_validation.append(res_file.file_name)
+            res_file.delete()
+            err_msg = base_err_msg.format(reason="single file upload must be a zip file",
+                                          files=", ".join(files_failed_validation))
```

Comment:
@pkdash In this special case, I think ```files_failed_validation``` list should only have one item, so the ```join``` call is not needed.